metadatastore/mds.py

```diff
@@ -17,9 +17,17 @@ def __init__(self, config, version=1):
         self._RUNSTART_CACHE = {}
         self._RUNSTOP_CACHE = {}
         self._DESCRIPTOR_CACHE = {}
-
+        self.reset_connection()
```

Comment:
I am not a huge fan of this because now it is hard to tell what state is in this object but eh